### ISSUE TITLE
0.1 release roadmap

### ISSUE THREAD

**Issue by fdecampredon** [TypeStrong/typescript-project-services#1]
Ok now that the project is a little stable I would like to create a little roadmap.

## Name
I quite hate the name that I have given to this project, if anybody has a better idea ... :D 

## Build system
* I don't know why I have gone with Make, but it's a bit boring and the project would seriously benefit from a better build system. (grunt + grunt-ts ? )
* the [script](https://github.com/TypeStrong/typescript-project-services/tree/master/scripts) generating the declaration file is interesting perhaps should we extract it in a separate project ?
* Integration with a travice-ci or something like that could be a big bonus

## Promise
Actually I have hidden the dependencies over 'bluebird' (see : https://github.com/TypeStrong/typescript-project-services/blob/master/src/main/project.ts#L20-L21) to avoid forcing people consuming this project to have a bluebird declaration file, however in fact it would be better to let people inject the promise library they are comfortable with.

## Documentation
The project need a read me a description and any kind of documentation helping to consume this project.

## Feature
Actually the project expose 4 services, completion, definition, error, formatting. The project need to expose pretty much all the possibility given by the typescript language service.

## Service interface
The project services interface is pretty much all the time consuming emitting position based on line/column (see: https://github.com/TypeStrong/typescript-project-services/blob/master/src/main/index.ts#L10-L13), it's pretty handy for brackets-typescript, since it's the format used for codemirror, however will it be the case for atom-typescript ?

## Tests
Some part of the project are somewhat well tested (like project/projectManager) but some other are not tested at all, since the project want to become a foundation for other project it needs to be well-tested and we need code-coverage

## Contributing
Finally we need a *contributing.md'*+ code guideline

Feel free to comment to give me your point of view, I'll edit the issue if I missed something.

**Comment by fdecampredon** [comment]
@basarat  I would like to publish a 0.1alpha : 
* Do you have any idea fir the name ? 'typescript-project-services' is horrible
* Is there a special TypeStrong npm account ?

**Comment by basarat** [comment]
> Is there a special TypeStrong npm account

No. Publish as yourself. You can add others over time using e.g. `npm owner add basarat` from the folder that contains `package.json`. 

> Do you have any idea for the name ? 'typescript-project-services' is horrible 

https://www.npmjs.com/package/tsps (typescript project services) pronounced `tspS` (http://youtu.be/4Tqqui0vEMw?t=45s)

**Comment by fdecampredon** [comment]
@basarat I'm pretty much ready for initial release, could you give a look at the project (doc, code, package.json, etc...) before publishing ?

**Comment by basarat** [comment]
Great stuff! love the docs!

**Comment by basarat** [comment]
publish away :heart:
